File: server/server_logging/telegram.py
import os
import requests
import logging
from dotenv import load_dotenv
load_dotenv()

TOKEN = os.environ.get("TELEGRAM_KEY")
CHANNEL_ID = os.environ.get("CHANNEL_ID")

if not TOKEN:
    raise ValueError("TELEGRAM_KEY environment variable is not set")

if not CHANNEL_ID:
    raise ValueError("CHANNEL_ID environment variable is not set")

TELEGRAM_API_URL = f"https://api.telegram.org/bot{TOKEN}"

class TelegramHandler(logging.Handler):

    def __init__(self):
        super().__init__()

    def emit(self, record):
        """
        Emit a record.

        This method sends a log record to a Telegram channel.

        Parameters:
        record (logging.LogRecord): The log record to be sent.
        """
        record = self.format(record)
        try:
            res = requests.post(url=TELEGRAM_API_URL+"/sendMessage", data={"text": record, "chat_id": CHANNEL_ID})
            res.raise_for_status()
        except Exception as e:
            print(f"Failed to send message to Telegram: {e}")

File: server/utils/telegram.js
const TOKEN = process.env.TELEGRAM_KEY;
const CHANNEL_ID = process.env.CHANNEL_ID;

if (!TOKEN || !CHANNEL_ID) {
  throw new Error("Please provide TELEGRAM_KEY and CHANNEL_ID in .env file");
}

const TELEGRAM_API_URL = `https://api.telegram.org/bot${TOKEN}`;

const sendMessage = async (text) => {
  const url = `${TELEGRAM_API_URL}/sendMessage`;
  const body = {
    chat_id: CHANNEL_ID,
    text,
  };

  const response = await fetch(url, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(body),
  });

  return await response.json();
};

module.exports = sendMessage;
